Retrofit doesn't work with runDistributable

**Problem:**

`Retrofit` works with `run` but getting `Caused by: java.lang.ClassNotFoundException: java.util.logging.Logger` while executing `runDistributable`

**Steps to reproduce**
1. Create a new project using IntelliJ IDEA
2. Add retrofit to your dependencies
```kotlin
val retrofitVersion = "2.9.0"
implementation("com.squareup.retrofit2:retrofit:$retrofitVersion")
``` 
3. Create an `OkHttpClient` anywhere in the code
4. Execute `runDistributable`

**Expected Result**
- Run the program without any error

**Actual Result**
- Programming crashes with `Caused by: java.lang.ClassNotFoundException: java.util.logging.Logger`

<details>
    <summary>Stacktrace</summary>

```
> Task :runDistributable
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: java/util/logging/Logger
        at okhttp3.internal.platform.Platform.<clinit>(Platform.java:82)
        at okhttp3.OkHttpClient.newSslSocketFactory(OkHttpClient.java:263)
        at okhttp3.OkHttpClient.<init>(OkHttpClient.java:229)
        at okhttp3.OkHttpClient$Builder.build(OkHttpClient.java:1015)
        at com.theapache64.stackzy.di.module.NetworkModule.provideOkHttpClient(NetworkModule.kt:52)
        at com.theapache64.stackzy.di.module.NetworkModule_ProvideOkHttpClientFactory.provideOkHttpClient(NetworkModule_ProvideOkHttpClientFactory.java:41)
        at com.theapache64.stackzy.di.module.NetworkModule_ProvideOkHttpClientFactory.get(NetworkModule_ProvideOkHttpClientFactory.java:31)
        at com.theapache64.stackzy.di.module.NetworkModule_ProvideOkHttpClientFactory.get(NetworkModule_ProvideOkHttpClientFactory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at com.theapache64.stackzy.di.module.NetworkModule_ProvideRetrofitFactory.get(NetworkModule_ProvideRetrofitFactory.java:31)
        at com.theapache64.stackzy.di.module.NetworkModule_ProvideRetrofitFactory.get(NetworkModule_ProvideRetrofitFactory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at com.theapache64.stackzy.di.module.NetworkModule_ProvideApiInterfaceFactory.get(NetworkModule_ProvideApiInterfaceFactory.java:31)
        at com.theapache64.stackzy.di.module.NetworkModule_ProvideApiInterfaceFactory.get(NetworkModule_ProvideApiInterfaceFactory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at com.theapache64.stackzy.data.repo.CategoriesRepo_Factory.get(CategoriesRepo_Factory.java:25)
        at com.theapache64.stackzy.data.repo.CategoriesRepo_Factory.get(CategoriesRepo_Factory.java:8)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at com.theapache64.stackzy.di.DaggerAppComponent.splashViewModel(DaggerAppComponent.java:82)
        at com.theapache64.stackzy.di.DaggerAppComponent.injectSplashScreenComponent(DaggerAppComponent.java:141)
        at com.theapache64.stackzy.di.DaggerAppComponent.inject(DaggerAppComponent.java:122)
        at com.theapache64.stackzy.ui.feature.splash.SplashScreenComponent.<init>(SplashScreenComponent.kt:22)
        at com.theapache64.stackzy.ui.navigation.NavHostComponent.createScreenComponent(NavHostComponent.kt:81)
        at com.theapache64.stackzy.ui.navigation.NavHostComponent.access$createScreenComponent(NavHostComponent.kt:25)
        at com.theapache64.stackzy.ui.navigation.NavHostComponent$router$1.invoke(NavHostComponent.kt:73)
        at com.theapache64.stackzy.ui.navigation.NavHostComponent$router$1.invoke(NavHostComponent.kt:73)
        at com.arkivanov.decompose.router.RouterEntryFactoryImpl.invoke(RouterEntryFactoryImpl.kt:32)
        at com.arkivanov.decompose.router.RouterEntryFactory$DefaultImpls.invoke$default(RouterEntryFactory.kt:11)
        at com.arkivanov.decompose.router.StackHolderImpl.initialStack(StackHolderImpl.kt:56)
        at com.arkivanov.decompose.router.StackHolderImpl.<init>(StackHolderImpl.kt:34)
        at com.arkivanov.decompose.DefaultRouterFactory.router(DefaultRouterFactory.kt:36)
        at com.arkivanov.decompose.DefaultComponentContext.router(DefaultComponentContext.kt)
        at com.theapache64.stackzy.ui.navigation.NavHostComponent.router(NavHostComponent.kt)
        at com.theapache64.stackzy.ui.navigation.NavHostComponent.<init>(NavHostComponent.kt:165)
        at com.theapache64.stackzy.ui.feature.ComposableSingletons$MainActivityKt$lambda-1$1$1.invoke(MainActivity.kt:33)
        at com.theapache64.stackzy.ui.feature.ComposableSingletons$MainActivityKt$lambda-1$1$1.invoke(MainActivity.kt:33)
        at com.arkivanov.decompose.extensions.compose.jetbrains.RootComponentBuilderKt.rootComponent(RootComponentBuilder.kt:37)
        at com.theapache64.stackzy.ui.feature.ComposableSingletons$MainActivityKt$lambda-1$1.invoke(MainActivity.kt:33)
        at com.theapache64.stackzy.ui.feature.ComposableSingletons$MainActivityKt$lambda-1$1.invoke(MainActivity.kt:31)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at com.theapache64.stackzy.ui.theme.ThemeKt$StackzyTheme$1$1.invoke(Theme.kt:45)
        at com.theapache64.stackzy.ui.theme.ThemeKt$StackzyTheme$1$1.invoke(Theme.kt:38)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.material.SurfaceKt$Surface$1.invoke(Surface.kt:116)
        at androidx.compose.material.SurfaceKt$Surface$1.invoke(Surface.kt:105)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:190)
        at androidx.compose.material.SurfaceKt.Surface-F-jzlyU(Surface.kt:102)
        at com.theapache64.stackzy.ui.theme.ThemeKt$StackzyTheme$1.invoke(Theme.kt:38)
        at com.theapache64.stackzy.ui.theme.ThemeKt$StackzyTheme$1.invoke(Theme.kt:37)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:190)
        at androidx.compose.material.TextKt.ProvideTextStyle(Text.kt:246)
        at androidx.compose.material.MaterialThemeKt$MaterialTheme$1.invoke(MaterialTheme.kt:80)
        at androidx.compose.material.MaterialThemeKt$MaterialTheme$1.invoke(MaterialTheme.kt:79)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:190)
        at androidx.compose.material.MaterialThemeKt.MaterialTheme(MaterialTheme.kt:71)
        at com.theapache64.stackzy.ui.theme.ThemeKt.StackzyTheme(Theme.kt:34)
        at com.theapache64.stackzy.ui.feature.ComposableSingletons$MainActivityKt$lambda-2$1.invoke(MainActivity.kt:29)
        at com.theapache64.stackzy.ui.feature.ComposableSingletons$MainActivityKt$lambda-2$1.invoke(MainActivity.kt:28)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.desktop.AppWindowKt$Window$1$1.invoke(AppWindow.kt:91)
        at androidx.compose.desktop.AppWindowKt$Window$1$1.invoke(AppWindow.kt:90)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.desktop.AppWindow$show$1.invoke(AppWindow.kt:439)
        at androidx.compose.desktop.AppWindow$show$1.invoke(AppWindow.kt:437)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:190)
        at androidx.compose.desktop.AppWindow$onCreate$1.invoke(AppWindow.kt:412)
        at androidx.compose.desktop.AppWindow$onCreate$1.invoke(AppWindow.kt:411)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.ui.platform.WrapperKt$setContent$1$1.invoke(Wrapper.kt:41)
        at androidx.compose.ui.platform.WrapperKt$setContent$1$1.invoke(Wrapper.kt:40)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:190)
        at androidx.compose.ui.platform.CompositionLocalsKt.ProvideCommonCompositionLocals(CompositionLocals.kt:121)
        at androidx.compose.ui.platform.WrapperKt$ProvideDesktopAmbients$1.invoke(Wrapper.kt:60)
        at androidx.compose.ui.platform.WrapperKt$ProvideDesktopAmbients$1.invoke(Wrapper.kt:59)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:190)
        at androidx.compose.ui.platform.WrapperKt.ProvideDesktopAmbients(Wrapper.kt:56)
        at androidx.compose.ui.platform.WrapperKt.access$ProvideDesktopAmbients(Wrapper.kt:1)
        at androidx.compose.ui.platform.WrapperKt$setContent$1.invoke(Wrapper.kt:40)
        at androidx.compose.ui.platform.WrapperKt$setContent$1.invoke(Wrapper.kt:39)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:121)
        at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.kt:46)
        at androidx.compose.runtime.ComposerKt.invokeComposable(Composer.kt:3415)
        at androidx.compose.runtime.ComposerImpl.composeContent$runtime(Composer.kt:2602)
        at androidx.compose.runtime.CompositionImpl.composeContent(Composition.kt:240)
        at androidx.compose.runtime.Recomposer.composeInitial$runtime(Recomposer.kt:524)
        at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:236)
        at androidx.compose.ui.platform.WrapperKt.setContent(Wrapper.kt:39)
        at androidx.compose.desktop.ComposeLayer.initOwner(ComposeLayer.kt:256)
        at androidx.compose.desktop.ComposeLayer.access$initOwner(ComposeLayer.kt:50)
        at androidx.compose.desktop.ComposeLayer$Wrapped.init(ComposeLayer.kt:83)
        at org.jetbrains.skiko.HardwareLayer.checkIsShowing(HardwareLayer.kt:30)
        at org.jetbrains.skiko.HardwareLayer.access$checkIsShowing(HardwareLayer.kt:7)
        at org.jetbrains.skiko.HardwareLayer$1.hierarchyChanged(HardwareLayer.kt:22)
        at java.desktop/java.awt.Component.processHierarchyEvent(Unknown Source)
        at java.desktop/java.awt.Component.processEvent(Unknown Source)
        at java.desktop/java.awt.Component.dispatchEventImpl(Unknown Source)
        at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
        at java.desktop/java.awt.Component.createHierarchyEvents(Unknown Source)
        at java.desktop/java.awt.Container.createHierarchyEvents(Unknown Source)
        at java.desktop/java.awt.Container.createHierarchyEvents(Unknown Source)
        at java.desktop/java.awt.Container.createHierarchyEvents(Unknown Source)
        at java.desktop/java.awt.Container.createHierarchyEvents(Unknown Source)
        at java.desktop/java.awt.Container.createHierarchyEvents(Unknown Source)
        at java.desktop/java.awt.Component.show(Unknown Source)
        at java.desktop/java.awt.Window.show(Unknown Source)
        at java.desktop/java.awt.Component.show(Unknown Source)
        at java.desktop/java.awt.Component.setVisible(Unknown Source)
        at java.desktop/java.awt.Window.setVisible(Unknown Source)
        at androidx.compose.desktop.ComposeWindow.setVisible(ComposeWindow.kt:86)
        at androidx.compose.desktop.AppWindow.show(AppWindow.kt:442)
        at androidx.compose.desktop.AppWindow.show$default(AppWindow.kt:428)
        at androidx.compose.desktop.AppWindowKt$Window$1.run(AppWindow.kt:90)
        at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
        at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
        at java.base/java.security.AccessController.doPrivileged(Unknown Source)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: java.lang.ClassNotFoundException: java.util.logging.Logger
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        ... 142 more
```
</details>


**Additional Informations**
```
IntelliJ IDEA 2020.3.2 (Community Edition)
Build #IC-203.7148.57, built on January 26, 2021
Runtime version: 11.0.9.1+11-b1145.77 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 4.15.0-133-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1934M
Cores: 6
Non-Bundled Plugins: wu.seal.tool.jsontokotlin, org.jetbrains.kotlin, il.co.galex.namethatcolor
Current Desktop: Unity
```

- **Real Project**:  https://github.com/theapache64/stackzy

--- a/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/dsl/NativeDistributions.kt
+++ b/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/dsl/NativeDistributions.kt
@@ -21,7 +21,7 @@ open class NativeDistributions @Inject constructor(
         set(layout.buildDirectory.dir("compose/binaries"))
     }
 
-    var modules = arrayListOf("java.desktop")
+    var modules = arrayListOf("java.desktop", "java.logging")
     fun modules(vararg modules: String) {
         this.modules.addAll(modules.toList())
     }

--- a/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/DesktopApplicationTest.kt
+++ b/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/DesktopApplicationTest.kt
@@ -113,4 +113,12 @@ class DesktopApplicationTest : GradlePluginTestBase() {
             check.logContains("Called lib2#util()")
         }
     }
+
+    @Test
+    fun testJavaLogger() = with(testProject(TestProjects.javaLogger)) {
+        gradle(":runDistributable").build().checks { check ->
+            check.taskOutcome(":runDistributable", TaskOutcome.SUCCESS)
+            check.logContains("Compose Gradle plugin test log warning!")
+        }
+    }
 }

--- a/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/test/TestProjects.kt
+++ b/gradle-plugins/compose/src/test/kotlin/org/jetbrains/compose/test/TestProjects.kt
@@ -5,4 +5,5 @@ object TestProjects {
     const val mpp = "application/mpp"
     const val jvmKotlinDsl = "application/jvmKotlinDsl"
     const val moduleClashCli = "application/moduleClashCli"
+    const val javaLogger = "application/javaLogger"
 }

--- a/gradle-plugins/compose/src/test/test-projects/application/javaLogger/build.gradle
+++ b/gradle-plugins/compose/src/test/test-projects/application/javaLogger/build.gradle
@@ -1,0 +1,24 @@
+plugins {
+    id "org.jetbrains.kotlin.jvm"
+    id "org.jetbrains.compose"
+}
+
+repositories {
+    google()
+    mavenCentral()
+    jcenter()
+    maven {
+        url "https://maven.pkg.jetbrains.space/public/p/compose/dev"
+    }
+}
+
+dependencies {
+    implementation "org.jetbrains.kotlin:kotlin-stdlib"
+    implementation compose.desktop.currentOs
+}
+
+compose.desktop {
+    application {
+        mainClass = "MainKt"
+    }
+}

--- a/gradle-plugins/compose/src/test/test-projects/application/javaLogger/settings.gradle
+++ b/gradle-plugins/compose/src/test/test-projects/application/javaLogger/settings.gradle
@@ -1,0 +1,11 @@
+pluginManagement {
+    plugins {
+        id 'org.jetbrains.kotlin.jvm' version 'KOTLIN_VERSION_PLACEHOLDER'
+        id 'org.jetbrains.compose' version 'COMPOSE_VERSION_PLACEHOLDER'
+    }
+    repositories {
+        mavenLocal()
+        gradlePluginPortal()
+    }
+}
+rootProject.name = "javaLogger"

--- a/gradle-plugins/compose/src/test/test-projects/application/javaLogger/src/main/kotlin/main.kt
+++ b/gradle-plugins/compose/src/test/test-projects/application/javaLogger/src/main/kotlin/main.kt
@@ -1,0 +1,3 @@
+fun main() {
+    java.util.logging.Logger.getAnonymousLogger().warning("Compose Gradle plugin test log warning!")
+}

